pgpool2_exporter.yaml

```diff
@@ -0,0 +1,108 @@
+package:
+  name: pgpool2_exporter
+  version: "1.2.2"
+  epoch: 0
+  description: Prometheus exporter for Pgpool-II metrics.
+  copyright:
+    - license: MIT
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: http://github.com/pgpool/pgpool2_exporter
+      tag: v${{package.version}}
+      expected-commit: 06a68dead344308c3346f53aba811c929eff71d5
+
+  - uses: go/build
+    with:
+      packages: ./cmd/pgpool2_exporter
+      modroot: .
+      output: pgpool2_exporter
+      ldflags: |
+        -X github.com/prometheus/common/version.Version=${{package.version}}
+        -X github.com/prometheus/common/version.Branch=master
+        -X github.com/prometheus/common/version.Revision=$(git rev-parse HEAD)
+        -X github.com/prometheus/common/version.BuildDate=$(date -u -d "@$([ "${SOURCE_DATE_EPOCH:-0}" -eq 0 ] && date +%s || echo $SOURCE_DATE_EPOCH)" +'%Y-%m-%dT%H:%M:%SZ')
+        -X github.com/prometheus/common/version.BuildUser=chainguard-dev
+
+subpackages:
+  - name: ${{package.name}}-entrypoint
+    pipeline:
+      - name: "Entrypoint setup logic for image"
+        runs: |
+          install -m777 ./docker-entrypoint.sh ${{targets.contextdir}}/entrypoint.sh
+
+update:
+  enabled: true
+  github:
+    identifier: pgpool/pgpool2_exporter
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - postgresql
+        - postgresql-client
+        - pgpool2
+        - curl
+        - bash
+        - shadow
+        - sudo-rs
+    environment:
+      PG_USER: "testuser"
+      PG_PASSWORD: "testpass"
+      PG_DB: "postgres"
+      PG_PORT: "5432"
+      PG_DATA: "/tmp/pgdata"
+      PGHOST: "localhost"
+      PGPOOL_PORT: "9999"
+      EXPORTER_PORT: "9187"
+  pipeline:
+    - name: "Test Version and help flags"
+      runs: |
+        pgpool2_exporter --version 2>&1 | grep -q "${{package.version}}"
+        pgpool2_exporter --help 2>&1 | grep -q "pgpool2_exporter"
+    - name: "Set up PostgreSQL"
+      runs: |
+        useradd -m $PG_USER
+        sudo -u $PG_USER initdb -D $PG_DATA --username=$PG_USER
+        # Start PostgreSQL in the background
+        sudo -u $PG_USER pg_ctl -D $PG_DATA -o "-p $PG_PORT" -w start
+        # Grant privileges to DB
+        psql -U $PG_USER -h "$PGHOST" -p "$PG_PORT" -d ${PG_DB} -c "GRANT ALL PRIVILEGES ON DATABASE $PG_DB TO $PG_USER;"
+    - name: "Start Pgpool-II"
```

Comment:
Tests look great, added couple of suggestions and would also suggest if we can use ` uses: test/daemon-check-output` pipeline. Thanks for working on it!

Reply:
Noted, `test/daemon-check-output` seems pretty good for checking operators logs. Since `pgpool2_exporter` metrics are scraped through a server endpoint rather than daemon logs, I've added more `grep` validations instead.

Thanks for reviewing the PR!

Reply:
Updated tests with `test/daemon-check-output`. Thanks @mritunjaysharma394